### ISSUE TITLE
--api: command not found

### ISSUE THREAD

**Issue by wyunj** [espnet/espnet#4307]
I train conformer model using librispeech recipe (my own data), but when in stage 5: decoding, I got this:
run.sh: line 312: --api: command not found

**Basic environments:**
 - OS information:  Linux 
 - python version: 3.7.4
 - espnet version: 0.10.5a1
 - pytorch version 1.6.0

**Task information:**
 - Task: ASR
 - Recipe:  librispeech
 - ESPnet1

**Comment by kan-bayashi** [comment]
Did you modify the run.sh by yourself?
Maybe unintented space is inserted after `\`

**Comment by wyunj** [comment]
Thanks very much. And I have another question, does gpu parallelism make a big difference? I train conformer model using my own data, everything esle is the same except the number of gpus. And the result, 
- ngpu=2, wer=20.1
- ngpu=1, wer=19.0

**Comment by sw005320** [comment]
Do you mean that you change the number of GPUs during training?
Our training is not deterministic, and such changes may happen, especially when that amount of training data is small.

**Comment by wyunj** [comment]
Thanks for your reply! yes, I found that the results varied greatly when I trained with different numbers of gpus. My training data is 300h.
![捕获](https://user-images.githubusercontent.com/99406640/167336325-3e81c584-5435-4ac1-963c-d7fbec0e6350.PNG)
Do you mean this function is not ensure pytorch produces deterministic results?

**Comment by sw005320** [comment]
This is already enabled, but randomness still exists.
Also, in espnet1, when we use multiple GPUs, the actual batch size is multiplied by the number of GPUs.
So, to get similar performance to the single GPU case, please make the batch size half.

**Comment by wyunj** [comment]
Thanks very much! But I find that even under the condition of the same effective batch size, there is also a big difference in performance (like India 19.9/20.7). Is this because training is not deterministic? Our training data is 300h (Indian, US and England, 100h respectively) from CommonVoice corpus. The test set including extra 6 hours of data from Singapore, Australia and Cananda (as unseen data).
![image](https://user-images.githubusercontent.com/99406640/168133938-8693d383-b311-4c35-9808-137d5bae44b4.png)

**Comment by wyunj** [comment]
In addition, I also run the model on librispeech100 (no speed disturb). Similar to CommonVoice, the smaller effective batch size is, the better the performance seems to be. However, Conformer encoder contains a lot of Batch normalization, so theoretically, the larger the batch size is, the better the performance will be. Now it seems that the result contradicts the theory. I wonder if you came to a similar conclusion on librispeech100?
Thanks in advance!
![image](https://user-images.githubusercontent.com/99406640/169331757-679d64a7-de5f-47a8-8e7f-1af3296f050c.png)

**Comment by sw005320** [comment]
Thanks for your report.
It is good to know that the similar "effective" batch size does not change the performance so much in this librispeech100 example (but it seems to happen in the other dataset...).
And I agree that too small batch size is harmful to the batch norm in conformer, but at the same time, too large batch size could make the optimization difficult for a small amount of training data. So, I often observed this phenomenon. 
(Overall, I'm regarding the "effective" batch size as a kind of optimization hyper-parameter and tuning it in some cases.
